src/System.Collections/tests/Generic/Dictionary/Dictionary.Tests.cs

```diff
@@ -250,6 +250,33 @@ public void Remove_NonExistentEntries_DoesNotPreventEnumeration()
             }
         }
 
+        [Fact]
+        public void Clear_OnEmptyCollection_DoesNotInvalidateEnumerator()
+        {
+            if (PlatformDetection.IsFullFramework)
```

Comment:
use ```[SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework)]```
Can use please also add the reason why we are skipping this test for .net core

Reply:
The implementation of `Clear` was changed recently to bring it in line with the behavior of `Remove` and not invalidate the enumerator. The `if()` clause here is wrong. 

`PlatformDetection` is definitely unnecessary here. Instead the test should probably rely on expected behavior. Moved test to `Dictionary_IDictionary_NonGeneric_Tests` and instead inspect `ModifyEnumeratorSucceeds` for expected behavior.